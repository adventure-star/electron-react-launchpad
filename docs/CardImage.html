<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: CardImage</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: CardImage</h1>

    




<section>

<header>
    
        <h2>CardImage</h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    <h4 class="name" id="CardImage"><span class="type-signature"></span>new CardImage<span class="signature">(props)</span><span class="type-signature"></span></h4>

    



<div class="description">
    The CardImage component, an optional part the Card component (MaterializeCSS project)
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>props</code></td>
            

            <td class="type">
            
                
<span class="param-type">*</span>


            
            </td>

            

            

            <td class="description last">React property object</td>
        </tr>

    
    </tbody>
</table>






    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>image</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The image URL</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>title</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The title to appear on the image</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="renderer_components_card_cardimage.jsx.html">renderer/components/card/cardimage.jsx</a>, <a href="renderer_components_card_cardimage.jsx.html#line17">line 17</a>
    </li></ul></dd>
    

    

    

    
</dl>















    <h5>Example</h5>
    
    <pre class="prettyprint"><code>An example can be found here: http://materializecss.com/cards.html
         import { Card, CardImage } from './components/card'
         &lt;Card>
            &lt;CardImage image='http://.../image.jpg' title='My image title' />
         &lt;/Card></code></pre>



    
    </div>

    

    

    

     

    

    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="BaseTextInput.html">BaseTextInput</a></li><li><a href="Card.html">Card</a></li><li><a href="CardAction.html">CardAction</a></li><li><a href="CardContent.html">CardContent</a></li><li><a href="CardImage.html">CardImage</a></li><li><a href="CardPanel.html">CardPanel</a></li><li><a href="CardStacked.html">CardStacked</a></li><li><a href="CardTitle.html">CardTitle</a></li><li><a href="Collection.html">Collection</a></li><li><a href="CollectionHeader.html">CollectionHeader</a></li><li><a href="CollectionItem.html">CollectionItem</a></li><li><a href="CollectionLink.html">CollectionLink</a></li><li><a href="EmailInput.html">EmailInput</a></li><li><a href="FixedButton.html">FixedButton</a></li><li><a href="FloatingButton.html">FloatingButton</a></li><li><a href="PasswordInput.html">PasswordInput</a></li><li><a href="RaisedButton.html">RaisedButton</a></li><li><a href="TextInput.html">TextInput</a></li><li><a href="WindowManager.html">WindowManager</a></li></ul><h3>Global</h3><ul><li><a href="global.html#getWindowManager">getWindowManager</a></li><li><a href="global.html#loadFile">loadFile</a></li><li><a href="global.html#openFile">openFile</a></li><li><a href="global.html#showOpenDialog">showOpenDialog</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.4.0</a> on Tue Jul 18 2017 21:53:28 GMT+0200 (CEST)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>