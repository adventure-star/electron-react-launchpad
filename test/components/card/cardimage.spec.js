import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import CardImage from '../../../src/renderer/components/card/cardimage.jsx'

function setupStandard() {
    const renderer = ReactTestRenderer.create(<CardImage image='testimage' title='title'>Test content</CardImage>);
    const component = renderer.toJSON();

    return {
        component
    };
}

describe('Card - Card image component', () => {
    it('should be of type span', () => {
        const { component } = setupStandard();
        expect(component.type).toBe('div');
    });

    it('should have only the class "card-image"', () => {
        const { component } = setupStandard();
        expect(component.props.className).toBe('card-image');
    });

    it('component should have 3 children', () => {
        const { component } = setupStandard();
        expect(component.children.length).toBe(3);
    });

    it('component should have an image with src', () => {
        const { component } = setupStandard();
        expect(component.children[0].props.src).toBe('testimage');
    });

    it('component should have an title', () => {
        const { component } = setupStandard();
        expect(component.children[1].children[0]).toBe('title');
    });

    it('component should have extra content', () => {
        const { component } = setupStandard();
        expect(component.children[2]).toBe('Test content');
    });

    it('Matches/Renders a snapshot', () => {
        const { component } = setupStandard();
        expect(component).toMatchSnapshot();
    });
});
