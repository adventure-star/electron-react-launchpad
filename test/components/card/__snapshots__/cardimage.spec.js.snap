// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card - Card image component Matches/Renders a snapshot 1`] = `
<div
  className="card-image"
>
  <img
    src="testimage"
  />
  <span
    className="card-title"
  >
    title
  </span>
  Test content
</div>
`;
